nanofi/ecu/tail_file.c

```
@@ -0,0 +1,214 @@
+#include "core/string_utils.h"
+#include "core/cstructs.h"
+#include "core/file_utils.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <limits.h>
+#include <signal.h>
+#include <sys/stat.h>
+
+struct flow_file_records * flowfiles = NULL;
+nifi_instance * instance = NULL;
+standalone_processor * proc = NULL;
+int file_offset = 0;
+int stopped = 0;
+flow_file_list ff_list;
+token_list tks;
+
+void signal_handler(int signum) {
+    if (signum == SIGINT || signum == SIGTERM) {
+        stopped = 1;
+    }
+}
+
+void transmit_flow_files(nifi_instance * instance) {
+    flow_file_list_node * head = ff_list.head;
+    while (head) {
+        transmit_flowfile(head->ff_record, instance);
+        head = head->next;
+    }
+}
+
+void set_offset(int offset) {
+    file_offset = offset;
+}
+
+int get_offset() {
+    return file_offset;
+}
+
+void on_trigger_callback(processor_session * ps, processor_context * ctx) {
+
+    char file_path[4096];
+    char delimiter[2];
+
+    if (get_property(ctx, "file_path", file_path, 50) != 0) {
+        return;
+    }
+
+    if (get_property(ctx, "delimiter", delimiter, 2) != 0) {
+        return;
+    }
+
+    if (strlen(delimiter) == 0) {
+        printf("Delimiter not specified or it is empty\n");
+        return;
+    }
+    char delim = '\0';
+    if (strlen(delimiter) > 0) {
+        delim = delimiter[0];
+    }
+
+    if (delim == '\0') {
+        printf("Invalid delimiter \n");
+        return;
+    }
+
+    if (delim == '\\') {
+          if (strlen(delimiter) > 1) {
+            switch (delimiter[1]) {
+              case 'r':
+                delim = '\r';
+                break;
+              case 't':
+                delim = '\t';
+                break;
+              case 'n':
+                delim = '\n';
+                break;
+              case '\\':
+                delim = '\\';
+                break;
+              default:
```

Returning an error here instead of silently using \ as the delimiter would be better, but I can live with it.